Assets/_PlatformerDevelopment/Scripts/Views/StartScreenBehaviour.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace PersonalDevelopment
+{
+    public class StartScreenBehaviour : MonoBehaviour
+    {
+        public event Action OnButtonPressed;
+        
+        [SerializeField] private Button _startButton = null;
+
+        public void Initialize()
+        {
+            _startButton.onClick.AddListener(() =>
+            {
+                if (OnButtonPressed != null)
+                {
+                    OnButtonPressed();
+                }
+            });
+        }
+
+        public void CleanUp()
+        {
+            _startButton.onClick.RemoveAllListeners();
```

Comment:
Maybe add nullcheck for the button?

Reply:
I'm curious, is it better to not null check serialized field, so we can catch errors faster?

Reply:
Hmm I see what you mean, we could make it a another optional coding standard? And we can see what the rest of the devs want to do?

Reply:
I'm just playing devils advocate, i just thought of that reason just now. What do you think?

Reply:
The principal is that if the game ever get big enough, and a certain variable is used enough, something (could be predicted or **not predicted**) could force the varaible to be null. So when doing something on it, a nullpointerexception could occur.

Reply:
In this situation, this is a serializedfield.

If it's data coming from somewhere else, I agree, checking if it's null is okay. 
But since this is a serializedField Component, I think that we don't need to check for null

Reply:
Hmm that makes sense. But there could be a scenario where a serializedfield variable (i.e lets say gameobject), could get destroyed from another method , which was not from intention (unlikely though possible I guess), so calling with null checking could be dangerous all the same

Reply:
But we would've caught that in PR.

 Something like that shouldn't be happening in the first place.

Alot of what ifs are caught by having more experienced programmers.

Reply:
And if it does get destroyed we wouldve caught it from the console?

Reply:
Really, the only side i like doing it without null check, is that it can be checked on console early on. and that's valuable